src/lib/server/cms.ts

```diff
@@ -29,6 +37,90 @@ purify.addHook('afterSanitizeAttributes', function (node) {
 		node.setAttribute('rel', 'noopener');
 	}
 });
+
+// Zod schema for external product data validation
+const externalProductSchema = z.object({
+	_id: z.string(),
+	name: z.string(),
+	shortDescription: z.string(),
+	price: z.object({
+		amount: z.number(),
+		currency: z.string()
+	}),
+	pictures: z.array(z.any()).optional()
+});
+
+async function fetchExternalProduct(url: string): Promise<ExternalProductData | null> {
+	try {
+		// Parse URL to extract base and slug
+		const urlObj = new URL(url);
+		const pathMatch = urlObj.pathname.match(/\/product\/([^\/]+)/);
+
+		if (!pathMatch) {
+			console.error('Invalid product URL format:', url);
+			return null;
+		}
+
+		const slug = pathMatch[1];
+		const apiUrl = `${urlObj.origin}/.well-known/product/${slug}`;
+
+		const response = await fetch(apiUrl, {
+			headers: { Accept: 'application/json' },
+			signal: AbortSignal.timeout(5000) // 5s timeout
+		});
+
+		if (!response.ok) {
+			console.error(`Failed to fetch external product: ${response.status}`);
+			return null;
+		}
```

Comment:
Returning `null` will simply ignore the widget exists. It would be more useful to display to the user that a remote product was supposed to be displayed here, but could not be fetch for _reason_.